Tests/TuistCacheTests/GraphMappers/CacheTreeShakingGraphMapperTests.swift

```diff
@@ -157,4 +159,53 @@ final class CacheTreeShakingGraphMapperTests: TuistUnitTestCase {
         // Then
         XCTAssertEmpty(gotGraph.workspace.schemes)
     }
+
+    func test_map_removes_pruned_targets_from_scheme() throws {
+        // Given
+        let path = AbsolutePath("/project")
+        let targets = [
+            Target.test(name: "first", prune: true),
+            Target.test(name: "second", prune: false),
+            Target.test(name: "third", prune: true),
+        ]
+        let scheme = Scheme.test(
+            name: "Scheme",
+            buildAction: .test(targets: targets.map { TargetReference(projectPath: path, name: $0.name) }),
+            testAction: .test(
+                targets: targets.map { TestableTarget(target: TargetReference(projectPath: path, name: $0.name)) },
+                coverage: true,
+                codeCoverageTargets: targets.map { TargetReference(projectPath: path, name: $0.name) }
+            )
+        )
+        let project = Project.test(path: path, targets: targets, schemes: [scheme])
+
+        let graph = ValueGraph.test(
+            path: project.path,
+            projects: [project.path: project],
+            targets: [project.path: Dictionary(uniqueKeysWithValues: targets.map { ($0.name, $0) })],
+            dependencies: [:]
+        )
+
+        // When
+        let (gotGraph, _) = try subject.map(graph: graph)
+
+        // Then
+        let gotProject = try XCTUnwrap(gotGraph.projects.first?.value)
+        let gotScheme = try XCTUnwrap(gotProject.schemes.first)
+
+        XCTAssertEqual(gotScheme.buildAction?.targets.count, 1)
```

Comment:
I tend to do:
```swift
let expectedGraph = ValueGraph(...all the expected properties...)
```

It makes the test more exhaustive than checking for individual properties IMHO (you can also omit all the individual checks this way)

Reply:
I like what you suggest! Updating